cycode/cli/code_scanner.py

```diff
@@ -328,18 +330,25 @@ def scan_disk_files(context: click.Context, path: str, files_to_scan: List[str])
 
     is_git_diff = False
 
-    documents: List[Document] = []
-    for file in files_to_scan:
-        progress_bar.update(ProgressBarSection.PREPARE_LOCAL_FILES)
+    try:
+        documents: List[Document] = []
+        for file in files_to_scan:
+            progress_bar.update(ProgressBarSection.PREPARE_LOCAL_FILES)
 
-        content = get_file_content(file)
-        if not content:
-            continue
+            content = get_file_content(file)
+            if not content:
+                continue
+
+            documents.append(Document(file, content, is_git_diff))
```

Comment:
we already iterate by documents. we can add 

if _is_tfplan_document:
 ... do magic

the thing here instead of creating one more loop below

it will 100% match the logic that tf plans should be only in "scan path" 

it will look like this:
```python
if _is_iac(scan_type) and is_tfplan_document(file, content):
    documents.append(_handle_tflpan_document(...))
else:
    documents.append(Document(file, content, is_git_diff))
```

Reply:
yes, I'll ad something similar to this suggestion